enumerables.rb

```diff
@@ -19,47 +21,84 @@ def my_select
     res
   end
 
-  def my_all?
-    return enum_for(__callee__) unless block_given?
+  # rubocop:disable Metrics/PerceivedComplexity
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def my_all?(cond = nil)
+    if !block_given? && cond.nil?
+      my_each { |i| return false unless i }
+
+    elsif block_given?
+      my_each { |i| return false unless yield(i) }
+
+    elsif cond.is_a? Class
 
-    my_each { |i| return false unless yield(i) }
+      my_each { |i| return false unless i.is_a? cond }
+    elsif cond.is_a? Regexp
+
+      my_each { |i| return false unless i.match cond }
+    else
+      my_each { |i| return false unless i == cond }
+    end
     true
   end
 
-  def my_any?
-    return enum_for(__callee__) unless block_given?
+  def my_any?(cond = nil)
+    if !block_given? && cond.nil?
+      my_each { |i| return true if i }
+
+    elsif block_given?
+      my_each { |i| return true if yield(i) }
+
+    elsif cond.is_a? Class
 
-    my_each { |i| return true if yield(i) }
+      my_each { |i| return true if i.is_a? cond }
+    elsif cond.is_a? Regexp
+
+      my_each { |i| return true if i.match cond }
+    else
+      my_each { |i| return true if i == cond }
+    end
     false
   end
 
-  def my_none?
-    return enum_for(__callee__) unless block_given?
+  # rubocop:enable Metrics/PerceivedComplexity
+  # rubocop:enable Metrics/CyclomaticComplexity
 
-    my_each { |i| return false if yield(i) }
-    true
+  def my_none?(cond = nil)
+    if !block_given? && cond.nil?
+      !my_any?
+
+    elsif block_given?
+
+      !my_any? { yield }
+
+    else
+      !my_any?(cond)
+    end
   end
 
+  #  def my_none?(cond)
+  #    if block_given?
+  #
+  #      my_each { |i| return false if yield(i) }
+  #
+  #    else
+  #      my_each { |i| return false if (i.is_a? cond) || (i == cond) }
+  #    end
+  #    true
+  #  end
+
   def my_count(cond = nil)
     c = 0
     if block_given?
 
       my_each { |i| c += 1 if yield(i) }
-    else
+    elsif !cond.nil?
       my_each { |i| c += 1 if i == cond }
-
-    end
-    c
-  end
-
-  def my_map(&oper)
-    res = []
-    if block_given?
-      my_each { |i| res << yield(i) }
     else
-      my_each { |i| res << oper.call(i) }
+      c = size
     end
-    res
+    c
   end
 
   def my_inject(acumulator = nil, operator = nil)
```

Comment:
`my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- raises a "LocalJumpError" when no block or argument is given. But the current configuration does not cover it.

Reply:
Added the raise local jump error